README.md

```diff
@@ -82,3 +68,59 @@ viewModel.py have to do items, doing and done items
 unit test for checking to do items-paased test
 Integration Tests 
 End to End Selenium Tests
+
+Module-4 Vagrant
+
+
+Step-1 Downloaded and Installed VirtualBox
+Step-2 Downloaded and Installed Vagrant
+Visual Studio Code
+To start run vagrant init
+Vagrantfile and .vagrant are created
+Add .vagrant sub directory to .gitignore
+Vagrantfile" in the root of repository.
+Choose a base box : hashicorp/bionic64
+This file specifies the box we're using
+A box is a virtual machine image thatis starting point.
+ hashicorp/bionic64 is the image name
+ Launch VM-- $ vagrant up
+ connect to iVM using a secure shell (SSH): $ vagrant ssh
+ 
+ run python --version and
+python3 --version to discover the system Python versions pre-installed on the box.
+ 
+To manage multiple python versions on a single machine Python environment, use pyenv
+ Install pyenv prerequisites
+Install pyenv
+
+Download a new Python version using 
+pyenv install 3.9.0
+To make  new Python version the default one
+pyenv global 3.9.0 
+Re-provision  VM using 
+vagrant provision
+
+download and install poetry by adding the following command to
+ provisioning script:
+curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+Navigate to app code: 
+cd /vagrant
+Install dependencies:
+ poetry install
+Launch flask's development server: 
+poetry run flask run
+
+Trouble shooting
+delete .venv
+Edit poetry.toml file
+[virtualenvs]
+create = true
+in-project = false
+
+Write a trigger to install dependencies
+
+Add Network to vagrant file
+config.vm.network "forwarded_port", guest: 5000, host: 5000
```

Comment:
Worth noting that most of this is basically duplicating the Vagrantfile itself - this can be useful when explaining it to yourself, but tends to end up out of date quite quickly, and isn't of use to another developer who just wants to know how to run the app in Vagrant, i.e. which tools to download and what command to run